src/main/java/org/opensearch/security/tools/democonfig/SecuritySettingsConfigurer.java

```diff
@@ -199,8 +201,9 @@ void updateAdminPassword() throws IOException {
      */
     private boolean isAdminPasswordSetToAdmin(String internalUsersFile) throws IOException {
         JsonNode internalUsers = YAML_MAPPER.readTree(new FileInputStream(internalUsersFile));
+        PasswordHasher passwordHasher = new BCryptPasswordHasher();
         return internalUsers.has("admin")
-            && OpenBSDBCrypt.checkPassword(internalUsers.get("admin").get("hash").asText(), DEFAULT_ADMIN_PASSWORD.toCharArray());
+            && passwordHasher.check(DEFAULT_ADMIN_PASSWORD.toCharArray(), internalUsers.get("admin").get("hash").asText());
```

Comment:
I'm curious as to why the order is swapped around here, is that the standard/could we keep it consistent with the OpenBSDBCrypt parameter ordering?

Reply:
Don't think there is a standard as I've seen them ordered either way depending on the library. I've put the password parameter first to keep it consistent with the verification function in Password4j (and to me it seems more intuitive) however I can certainly flip them around if required!